### PR TITLE
Event sourced object to app config and migration command

### PR DESCRIPTION
**Pull request by zgelp**
Edited documentation and provided config with passing migrations.

### CODE REVIEW

**Comment by johnbywater** [comment]
Thank you!

File: README.md [hunk 1]
```diff
@@ -92,14 +92,37 @@ in the application's environment.
...
+    default_auto_field = "django.db.models.BigAutoField"
+    name = "<project_name>.training_school"
+
+    def ready(self):
+        call_command("migrate")
+        self.create_training_school()
+
+    def create_training_school(self):
+        training_school = TrainingSchool(
+            env={"PERSISTENCE_MODULE": "eventsourcing_django"}
```
Comment (Wintersina) [inline README.md:116]: just one notes...
we ended up putting this in a settings file.
`# Event Sourcing
EVENT_SOURCING_SETTINGS = {
    "env": {"PERSISTENCE_MODULE": "eventsourcing_django", "IS_SNAPSHOTTING_ENABLED": "y", "DJANGO_DB_ALIAS": "default"}
}`

then in our

utility where we wanna instantiate the class, we dependency inject.
`class BillingUtility():
    def __init__(self):
        self.billing_event_application = BillingEventApplication(**settings.EVENT_SOURCING_SETTINGS)`

thank you for sharing these notes too!
wanted to provide our example as well.

Comment (johnbywater) [reply README.md:116]: That's neat!

**Comment by johnbywater** [comment]
@zgelp @Wintersina: Just wanted to mention that I have folded in both of your contributions here: https://github.com/pyeventsourcing/eventsourcing-django?tab=readme-ov-file#event-sourcing-in-django

**Comment by johnbywater** [comment]
@zgelp @Wintersina: Please note (1) I changed the Application class so it doesn't make a call to the database when it is constructed, which removes the dependency on the database migration having happened before the application object is constructed, and (2) Django 5.1 has a warning about calls to the database from within the `ready()` method of an `AppConfig` object so it seems calling `migrate` from there isn't appropriate. Hence the example doesn't do this. Also the `self` argument is exactly the same object that you get from `apps.get_app_config()` so I changed that too. And the code @zgelp submitted actually failed the `test_docs()` test because the code in the README file is executed by the test suite, so I added a `training_school` Django app to make this example work. Also I enhanced the example from @Wintersina of defining the event sourcing application environment in the Django settings file by adding config for compression and encryption, to further demonstrate the power of having these things centralised in the settings file. Thanks again for your contributions, I really appreciate it.

**Pull request merged**